website/docs/r/cloudfunctions_function.html.markdown

```diff
@@ -0,0 +1,90 @@
+---
+layout: "google"
+page_title: "Google: google_cloudfunctions_function"
+sidebar_current: "docs-google-cloudfunctions-function"
+description: |-
+  Creates a new Cloud Function.
+---
+
+# google\_cloudfunctions\_function
+
+Creates a new Cloud Function. For more information see
+[the official documentation](https://cloud.google.com/functions/docs/)
+and
+[API](https://cloud.google.com/functions/docs/apis).
+
+## Example Usage
+
+```hcl
+resource "google_storage_bucket" "bucket" {
+  name = "test-bucket"
+}
+
+resource "google_storage_bucket_object" "archive" {
+  name   = "index.zip"
+  bucket = "${google_storage_bucket.bucket.name}"
+  source = "./path/to/zip/file/which/contains/code"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  name                  = "function-test"
+  description           = "My function"
+  available_memory_mb   = 128
+  source_archive_bucket = "${google_storage_bucket.bucket.name}"
+  source_archive_object = "${google_storage_bucket_object.archive.name}"
+  trigger_http          = true
+  timeout               = 60
+  entry_point           = "helloGET"
+  labels {
+	my-label = "my-label-value"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A unique name for the resource, required by GCE.
```

Comment:
This doesn't really involve GCE. The API docs have this as "A user-defined name of the function. Function names must be unique globally", which I think is a good description